import/webview/WebView.qml

```diff
@@ -31,19 +31,32 @@ RawWebView {
         webview.loadFrameScript("chrome://embedlite/content/embedhelper.js");
         webview.loadFrameScript("chrome://embedlite/content/SelectAsyncHelper.js");
         webview.addMessageListeners([
-            "embed:linkclicked",
-            "embed:filepicker",
-            "embed:selectasync",
-            "embed:alert",
-            "embed:confirm",
-            "embed:prompt",
-            "embed:auth",
-            "embed:permissions",
-            "embed:login",
-            "Content:ContextMenu"]);
+                                        "embed:linkclicked",
+                                    ]);
+    }
+
+    PickerOpener {
+        id: pickerOpener
+
+        property QtObject pageStackOwner: helper.findParentWithProperty(webView, "pageStack")
+
+        pageStack: pageStackOwner ? pageStackOwner.pageStack : undefined
+        contentItem: webview
+    }
+
+    PopupOpener {
+        id: popupOpener
+
+        pageStack: pickerOpener.pageStack
+        parentItem: webview
+        contentItem: webview
     }
 
     onRecvAsyncMessage: {
+        if (pickerOpener.handlesMessage(message) || popupOpener.handlesMessage(message)) {
```

Comment:
Did it like this, a bit inspired by QObject::event method:
```
if (pickerOpener.message(message, data) || popupOpener.message(message, data) {
    return;
}

```
Signal handler argument message is more like a topic and topic+data is the message. That's why message functions have prototype like ```message(topic, data)```